libr/include/r_util/r_logging.h

```diff
@@ -0,0 +1,71 @@
+#ifndef R_LOGGING_H
+#define R_LOGGING_H
+
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__GNUC__)
+#define MACRO_LOG_FUNC __FUNCTION__
+#define MACRO_WEAK_SYM
+// TODO: Windows weak symbols?
+#else
+#define MACRO_LOG_FUNC __func__
+#define MACRO_WEAK_SYM __attribute__ ((weak))
+#endif
+
+typedef void (*RLoggingFuncdef)(const char *output, const char *funcname,
+	const char *filename, unsigned int lineno, unsigned int level, const char *fmtstr, ...);
```

Comment:
Please add a `void *user` argument to the callback, we will need that at least in Cutter. Then also add a variable to contain the value that should be passed to it next to the places where a callback is stored.

Reply:
Whats the use case for this? Would it normally contain the RCore instance or?

Reply:
For the default one, probably. But for the task-local stuff I it will be important.